src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -470,11 +480,34 @@ public boolean isEnqueued() {
     }
 
     /**
-     * Clears this reference object and adds it to the queue with which
-     * it is registered, if any.
+     * Clears this reference object, then attempts to add it to the queue with
+     * which it is registered, if any.
+     *
+     * <p>If this reference is registered with a queue but not yet enqueued,
+     * the reference is added to the queue; this method is
+     * <b><i>successful</i></b> and returns true.
+     * If this reference is not registered with a queue, or was already enqueued
+     * (by the garbage collector, or a previous call to {@code enqueue}), this
+     * method is <b><i>unsuccessful</i></b> and returns false.
+     *
+     * <p><a href="{@docRoot}/java.base/java/lang/ref/package-summary.html#MemoryConsistency">Memory consistency effects</a>:
+     * Actions in a thread prior to calling
+     * {@code enqueue} <b><i>successfully</i></b>
+     * <a href="{@docRoot}/java.base/java/util/concurrent/package-summary.html#MemoryVisibility"><i>happen-before</i></a>
```

Comment:
Editorial. Without the markup, this says

> Actions in a thread prior to calling enqueue successfully happen-before the reference is removed...

This could be confusing, because "successfully" might be read to modify "happen-before". This raises questions such as "Is it possible for something to happen-before unsuccessfully?" Of course you want "successfully" to modify "enqueue" because you're relying on the definition of "successful" in the enqueue() spec. Suggest rewording:

> Actions in a thread prior to successful calls to `enqueue` _happen-before_ the reference is removed...

Reply:
Updated, though I made it singular ("a successful call to enqueue"), since we talk about "the" reference being removed.